packages/image_picker/image_picker_android/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -538,28 +546,34 @@ public boolean onRequestPermissionsResult(
   }
 
   @Override
-  public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
-    switch (requestCode) {
-      case REQUEST_CODE_CHOOSE_IMAGE_FROM_GALLERY:
-        handleChooseImageResult(resultCode, data);
-        break;
-      case REQUEST_CODE_CHOOSE_MULTI_IMAGE_FROM_GALLERY:
-        handleChooseMultiImageResult(resultCode, data);
-        break;
-      case REQUEST_CODE_TAKE_IMAGE_WITH_CAMERA:
-        handleCaptureImageResult(resultCode);
-        break;
-      case REQUEST_CODE_CHOOSE_VIDEO_FROM_GALLERY:
-        handleChooseVideoResult(resultCode, data);
-        break;
-      case REQUEST_CODE_TAKE_VIDEO_WITH_CAMERA:
-        handleCaptureVideoResult(resultCode);
-        break;
-      default:
-        return false;
-    }
+  public boolean onActivityResult(final int requestCode, final int resultCode, final Intent data) {
+    // Off-load result handling to a separate thread as it almost always involves I/O operations.
+    executor.execute(
+        () -> {
+          switch (requestCode) {
+            case REQUEST_CODE_CHOOSE_IMAGE_FROM_GALLERY:
+              handleChooseImageResult(resultCode, data);
+              break;
+            case REQUEST_CODE_CHOOSE_MULTI_IMAGE_FROM_GALLERY:
+              handleChooseMultiImageResult(resultCode, data);
+              break;
+            case REQUEST_CODE_TAKE_IMAGE_WITH_CAMERA:
+              handleCaptureImageResult(resultCode);
+              break;
+            case REQUEST_CODE_CHOOSE_VIDEO_FROM_GALLERY:
+              handleChooseVideoResult(resultCode, data);
+              break;
+            case REQUEST_CODE_TAKE_VIDEO_WITH_CAMERA:
+              handleCaptureVideoResult(resultCode);
+              break;
+          }
+        });
 
-    return true;
+    return requestCode == REQUEST_CODE_CHOOSE_IMAGE_FROM_GALLERY
```

Comment:
The duplication here is error-prone; couldn't we instead have a local runnable and in each `case` assign the runnable to a lamba that calls the `handle*` method, then after the switch `execute` it if it's non-null and then return true? That would avoid too much duplication within the switch, while not duplicating the handled list.

Reply:
You are right. I have updated the code with your suggestion.